### PR TITLE
numpy style docstrings

### PR DESCRIPTION
**Pull request by HarshaSatyavardhan**
Fixes #

Changes proposed in this pull request:

- added numpy style docstrings

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: dodiscover/replearning/gin.py [hunk 1]
```diff
@@ -51,8 +87,21 @@ def _extract_edgelists(adj_mat, names):
         return pdag
 
 
-    def fit(self, data, context):
+    def fit(self, data: 'DataFrame', context: 'DataFrame'):
```
Comment (robertness) [inline dodiscover/replearning/gin.py:90]: a context is an instance of the [Context class.](https://github.com/py-why/dodiscover/blob/main/dodiscover/context.py#L9)

File: dodiscover/replearning/gin.py [hunk 2]
```diff
@@ -51,8 +87,21 @@ def _extract_edgelists(adj_mat, names):
...
         """Fit to data.
+        
+        Parameters
+        ----------
+        data : DataFrame
+            The data to fit to.
+        context : DataFrame
```
Comment (robertness) [inline dodiscover/replearning/gin.py:97]: see comment above

**Review by robertness — APPROVED** [review]
I think this looks good.  Looking over the GIN code, I can't think of any reason to make this interface more complicated than it is.  Make that adjustment for the Context type hint and we can get this merged.

**Comment by robertness** [comment]
Also @HarshaSatyavardhan, this: https://github.com/py-why/dodiscover/issues/87

**Comment by adam2392** [comment]
Some CI failures:

- probably you need to merge in changes from `main` branch
- check https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md guide

**Comment by robertness** [comment]
Merging in to GIN PR

**Pull request merged**